### PR TITLE
Am 2820 hmc iac org roles

### PR DESCRIPTION
**Pull request by abul-syed**
### JIRA link (if applicable) ###

- [AM-2820](https://tools.hmcts.net/jira/browse/AM-2820) - New Organisational Role Assignment(IAC)

### Change description ###

Add HMC org roles for IAC

FTs Fix:

- #1541

> NB: This PR is has a dependency on hmcts/am-role-assignment-service#1922.  As a result **this PR has test only Preview-Chart and Jenkins changes that must be reversed prior to merger**.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by tomxelliott — DISMISSED** [review]
LGTM

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -76,7 +76,9 @@ env.PACT_BROKER_FULL_URL = "https://pact-broker.platform.hmcts.net"
 env.AMQP_HOST = "rd-servicebus-aat.servicebus.windows.net"
 env.CRD_TOPIC_NAME = "rd-caseworker-topic-aat"
 env.JRD_TOPIC_NAME = "rd-judicial-topic-aat"
-env.ROLE_ASSIGNMENT_URL = "http://am-role-assignment-service-aat.service.core-compute-aat.internal"
```
Comment (abul-syed) [inline Jenkinsfile_CNP:79]: Note: revert back to AAT RAS instance prior to master merge

File: charts/am-org-role-mapping-service/values.preview.template.yaml [hunk 2]
```diff
@@ -30,7 +30,9 @@ java:
     S2S_URL: http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
     CASE_WORKER_REF_APP_URL: http://rd-caseworker-ref-api-aat.service.core-compute-aat.internal
     JUDICIAL_REF_APP_URL: http://rd-judicial-api-aat.service.core-compute-aat.internal
-    ROLE_ASSIGNMENT_APP_URL: http://am-role-assignment-service-aat.service.core-compute-aat.internal
```
Comment (abul-syed) [inline charts/am-org-role-mapping-service/values.preview.template.yaml:33]: Note: revert back to AAT RAS instance prior to master merge

File: src/main/resources/validationrules/core/hearing-role-caseworker-global.drl [hunk 3]
```diff
@@ -29,7 +29,8 @@ import function uk.gov.hmcts.reform.orgrolemapping.util.UtilityFunctions.getJuri
    $up: CaseWorkerAccessProfile((roleId in ("1","2","16") && serviceCode in ("BBA3"))
                                   || (roleId in ("1","2") && serviceCode in ("ABA3","ABA5"))
                                   || (roleId in ("1","2") && serviceCode in ("AAA6","AAA7"))
-                                  || (roleId in ("1","2") && serviceCode in ("BHA1")),
+                                  || (roleId in ("1","2") && serviceCode in ("BHA1"))
+                                  || (roleId in ("1","2") && serviceCode in ("BFA1")),
```
Comment (prashanthkumarnb) [inline src/main/resources/validationrules/core/hearing-role-caseworker-global.drl:33]: can this be coded like this 
|| (roleId in ("1","2") && serviceCode in ("BHA1,"BFA1""))

Comment (prashanthkumarnb) [reply src/main/resources/validationrules/core/hearing-role-caseworker-global.drl:33]: Same for rest of the below lines

Comment (abul-syed) [reply src/main/resources/validationrules/core/hearing-role-caseworker-global.drl:33]: I have separated them out as the service codes are for different services `(BHA1 - ET, BFA1 - IA)`. Whereas the one above `(roleId in ("1","2") && serviceCode in ("AAA6","AAA7"))` the 2 service codes belong to the same CIVIL jurisdiction.

Comment (mattnayler) [reply src/main/resources/validationrules/core/hearing-role-caseworker-global.drl:33]: I agree with @abul-syed that it is useful to keep the service config on separate lines.

**Review by prashanthkumarnb — DISMISSED** [review]

File: src/functionalTest/resources/features/F-001/S-010.td.json [hunk 4]
```diff
@@ -34,6 +34,40 @@
             "substantive": "N"
           }
         },
+        {
```
Comment (mattnayler) [inline src/functionalTest/resources/features/F-001/S-010.td.json:37]: can we add sort criteria to the start of the array: 
```json 
{
        "__elementId__": "roleName"
},
```

File: src/functionalTest/resources/features/F-003/S-022.td.json [hunk 5]
```diff
@@ -39,6 +39,25 @@
             "workTypes" : "[[ANYTHING_PRESENT]]"
           },
           "authorisations" : "[[ANYTHING_PRESENT]]"
+        },
```
Comment (mattnayler) [inline src/functionalTest/resources/features/F-003/S-022.td.json:42]: can we add sort criteria to the start of the array: 
```json 
{
        "__elementId__": "roleName"
},
```

**Review by mattnayler — COMMENTED** [review]
As these changes are not behind a flag we need confirmation from other common services that is ok for these roles to be enabled for IAC prior to sing off of SIT.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolHearingOfficeOrgRoleMappingTest.java [hunk 6]
```diff
@@ -164,10 +174,13 @@ void shouldReturnHearingManagerAndViewerCaseWorker_Ctsc(String roleId, String se
         "1,AAA6,CIVIL",
         "2,AAA6,CIVIL",
         "1,BHA1,EMPLOYMENT",
-        "2,BHA1,EMPLOYMENT"
+        "2,BHA1,EMPLOYMENT",
+        "1,BFA1,IA"
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolHearingOfficeOrgRoleMappingTest.java]: For better coverage, yes. addressed in c94c50b

File: src/functionalTest/resources/features/F-001/S-006.td.json [hunk 7]
```diff
@@ -17,6 +17,40 @@
     "_extends_": "Common_200_Response",
     "body": {
       "roleAssignmentResponse" : [
+        {
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/functionalTest/resources/features/F-001/S-006.td.json:20]: addressed in c94c50b

File: src/functionalTest/resources/features/F-001/S-010.td.json (hunk 4 above)
Comment (abul-syed) [reply src/functionalTest/resources/features/F-001/S-010.td.json:40]: addressed in c94c50b

File: src/functionalTest/resources/features/F-003/S-022.td.json (hunk 5 above)
Comment (abul-syed) [reply src/functionalTest/resources/features/F-003/S-022.td.json:45]: addressed in c94c50b

File: src/functionalTest/resources/features/F-003/S-023.td.json [hunk 8]
```diff
@@ -38,6 +38,25 @@
           },
           "authorisations" : "[[ANYTHING_PRESENT]]"
         },
+        {
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/functionalTest/resources/features/F-003/S-023.td.json:44]: addressed in c94c50b

**Review by prashanthkumarnb — DISMISSED** [review]

**Review by mattnayler — DISMISSED** [review]
LGTM.

File: Jenkinsfile_CNP (hunk 1 above)
Comment (mattnayler) [reply Jenkinsfile_CNP]: Resolved in 929cbbdbda46f5be83a9db9d329ba1faa43eedbe.

File: charts/am-org-role-mapping-service/values.preview.template.yaml (hunk 2 above)
Comment (mattnayler) [reply charts/am-org-role-mapping-service/values.preview.template.yaml]: Resolved in 929cbbdbda46f5be83a9db9d329ba1faa43eedbe.

**Review by mattnayler — APPROVED** [review]
Re-approved after 929cbbdbda46f5be83a9db9d329ba1faa43eedbe.

**Review by mikebrownccd — APPROVED** [review]
LGTM

**Pull request merged**